### PR TITLE
Add access token request & tests

### PR DESCRIPTION
**Pull request by adamdullenty**
Jira ticket: https://bloomon.atlassian.net/browse/LR-928

This PR adds the initial access token request that will be called by future operations (e.g. create shipment, cancel shipment).

To add via a follow up PR:

* Passing in the currently configured rails cache via the config
* Using the rails cache to cache the access token for 1hr (when used within the test suite, we'll create a dummy cache that is compatible with the `fetch` method)

### CODE REVIEW

File: .ruby-style.yml [hunk 1]
```diff
@@ -1081,3 +1081,10 @@ Rails/HelperInstanceVariable:
   Include:
     - app/helpers/**/*.rb
     - components/**/helpers/*.rb
+
+RSpec/NamedSubject:
+  # Discussion https://bloomandwild.slack.com/archives/C4TKPLB9A/p1662550913780879?thread_ts=1662542489.995919&cid=C4TKPLB9A
+  Enabled: false
```
Comment (adamdullenty) [inline .ruby-style.yml:1087]: Not sure I agree with this being enabled, the syntax for naming a subject looks like this:

`subject(:user) { described_class.new(...) }`

then:

`expect(user).to...(something)`

which is confusing - what is `user`? I'd assume at first glance that it was a `user` record but it's not, it's an instance of the described class with some specific things passed into it.

Luckily we have decided to disable this in the monolith style rules so I've ported this over.

File: Gemfile [hunk 2]
```diff
@@ -4,3 +4,18 @@ source "https://rubygems.org"
...
+  gem "rake"
+  gem "rspec"
+  gem "rubocop"
+  gem "rubocop-performance"
+  gem "rubocop-rails"
+  gem "rubocop-rake"
+  gem "rubocop-rspec"
+  gem "vcr"
+  gem "webmock"
+end
```
Comment (adamdullenty) [inline Gemfile:21]: This was due to a Rubocop rule that says we should no longer specify development dependencies in the gemspec file - move them into the Gemfile instead.

File: lib/royal_mail_v4/access_token.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+    attr_reader :response, :options
+
+    def initialize(http_client: Faraday)
+      @http_client = http_client
+    end
+
+    def token
+      # TODO: Cache with Rails cache or default to a compatible memory store
+      @token ||= request_access_token
```
Comment (adamdullenty) [inline lib/royal_mail_v4/access_token.rb:13]: As per PR description, I'll add a follow up that allows us to pass in the currently configured cache from the Rails app, and within the test suite we'll just have a dummy cache that is compatible with the `fetch` method for read through caching.

File: royal_mail_v4.gemspec [hunk 4]
```diff
@@ -30,18 +30,5 @@ Gem::Specification.new do |spec|
...
-  spec.add_development_dependency "rake"
-  spec.add_development_dependency "rspec"
-  spec.add_development_dependency "rubocop"
-  spec.add_development_dependency "rubocop-performance"
-  spec.add_development_dependency "rubocop-rails"
-  spec.add_development_dependency "rubocop-rake"
-  spec.add_development_dependency "rubocop-rspec"
-  spec.add_development_dependency "vcr"
-  spec.add_development_dependency "webmock"
+  spec.add_dependency "faraday", "> 1.0", "< 2.0"
```
Comment (adamdullenty) [inline royal_mail_v4.gemspec:33]: For monolith compatibility

File: spec/support/fixtures/vcr_cassettes/request_access_token.yml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      - max-age=15724800; includeSubDomains
+      Cf-Cache-Status:
+      - DYNAMIC
+      Server:
+      - cloudflare
+      Cf-Ray:
+      - 7dd466ad7e64360d-MAN
+    body:
+      encoding: UTF-8
+      string: '{"access_token":"FILTERED","expires_in":3600,"token_type":"Bearer","scope":"mcssapi.all"}'
```
Comment (adamdullenty) [inline spec/support/fixtures/vcr_cassettes/request_access_token.yml:47]: Filtered token

File: spec/support/fixtures/vcr_cassettes/request_access_token.yml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+http_interactions:
+- request:
+    method: post
+    uri: https://authentication.proshipping.net/connect/token
+    body:
+      encoding: UTF-8
+      string: grant_type=client_credentials
+    headers:
+      Authorization:
+      - Basic <BASIC_AUTH_CREDENTIALS>
```
Comment (adamdullenty) [inline spec/support/fixtures/vcr_cassettes/request_access_token.yml:11]: Filtered basic auth

**Review by katherinlaine — APPROVED** [review]

**Pull request merged**